### DISCUSSION TITLE
Ability to run post-install anonymous scripts on unlocked packages

### DISCUSSION THREAD

**Discussion by wspringe** [forcedotcom/cli#2379] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, we are only able to run post-install anonymous scripts for 2GP managed packages.

**What are you trying to do**
Execute anonymous scripts after the install of an unlocked package, so we can automate some action like record creation after the package is installed. We want the setup and record creation for an unlocked package to be fully automated from an SFDX command so that we can avoid missing manual steps after an unlocked package installation.

**Describe the solution you'd like**
Ability to specify one or multiple anonymous scripts in the sfdx-project.json file of an SFDX project that is an unlocked package, and the script(s) will execute after the unlocked package is installed into a Salesforce org.

**Describe alternatives you've considered**
SFDX plugin, internal wrapper CLI solution, manually executing scripts after package installation

**Additional context**

[a comment by an automated account is not shown]

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-10346855](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000ZdHRUYA3/view)

**Comment by yippie** [comment]
Yes pease, this would be super helpful

**Comment by kyle-chiu** [comment]
This is a great idea!

**Comment by Slgrande** [comment]
We really need this!

**Comment by madhankumargi** [comment]
Much needed for automating CICD process

**Comment by pravirpal** [comment]
much needed, awaiting this feature.

**Comment by mshanemc** [comment]
> Much needed for automating CICD process

I'm curious about this.  If you've got a CI, why not `apex run` the code you want executed after the package installs?  [post-install scripts are very restricted for 2GP compared to what a user with a valid token to can do in apex via the CLI]

**Comment by VivekMChawla** [comment]
Thanks, everyone, for your feedback on this idea.

Unfortunately, we do not plan to deliver this feature as part of the Salesforce CLI. As @mshanemc pointed out, CI/CD pipelines that require post-install logic for unlocked packages should use `apex run`. 

As a product team, our focus is delivering functionality that enables outcomes for users that wouldn't otherwise be possible.  In other words, we want to give you a flexible and reliable set of building blocks and empower you to compose complex solutions with them.

If automatic execution of post-install logic with unlocked packages is still something you'd like, I'd suggest reaching out to the Packaging Team via the [Unlocked Packages group](https://trailhead.salesforce.com/trailblazer-community/groups/0F93A000000Lg5USAS?tab=discussion&sort=LAST_MODIFIED_DATE_DESC) in the Trailblazer Community.

Thank you!
